### PR TITLE
add file size test

### LINKED ISSUE #351: Add continuous integration test to this repository to prevent large files from being added
### Summary

We've just re-written history (#350) because our git repository was getting really big! 

It would be great if we could have travis test to see if there are any files larger than 5 MB in a pull request and have the tests fail if there are. (@KirstieJane can always use her master administrator powers to merge them in if they're really necessary).

### What needs to be done?

- [ ] Set up CI for this repository
- [ ] Write a test for file size in the PR. Fail if there is a file larger than 5 MB.

### Who can help?

* Anyone!

---

### Updates

*Lorem ipsum dolor sit amet, consectetur adipiscing.*

### PR DESCRIPTION
**Pull request by annakrystalli**
### Summary

Fixes #351 

The PR adds CI which runs a bash test script that fails if files larger than 5M are found in the repository. The script also allows adding files paths to files to be ignored during the test, allowing us to include larger files if required.

### List of changes proposed in this PR (pull-request)

* *Add `.travis.yml`* configured with `language: minimal` and to run a file checking test script
* *Add `scripts/test-file-size.sh* test script

### What should a reviewer concentrate their feedback on?

- [ ] * Feedback on Instructions to add to ignored files.*
 - should these instructions also be included somewhere else?
- [x] *Happy with the approach?.*
- [ ] *Happy with test sprint outs?*

### Acknowledging contributors

By submitting this pull request I confirm that:

- [ ] all contributors to this pull request who wish to be included are named in [contributors.md](https://github.com/alan-turing-institute/the-turing-way/blob/master/contributors.md).

@ptooley and @ptheywood also contributed to this PR

### CODE REVIEW

**Comment by annakrystalli** [comment]
So these tests, although only 12 secs to run seem to queue forever!

I wonder if it's anything to do with our configuration? I've not had my tests queued for this long on other projects. 🤷‍♀️

**Comment by ptooley** [comment]
Depends on how busy travis is. I think their free tier has a fixed worker pool and gets through jobs on a best effort basis.  I wouldn't worry too much since if there is an urgent PR you can always merge anyway if you are happy that local testing confirms nothing is broken.

**Comment by annakrystalli** [comment]
Also it looks like I can merge without a review. Do we need to reset branch protection?

**Comment by alexmorley** [comment]
> Also it looks like I can merge without a review. Do we need to reset branch protection?

yes. tagging @KirstieJane for that.

**Comment by alexmorley** [comment]
LGTM :tada:

File: scripts/test-file-size.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#! /bin/bash
+# add any files you want the test to ignore between the two ENDOFLIST placeholders.
```
Comment (alexmorley) [inline scripts/test-file-size.sh]: Maybe document that you have to assume that you are in the root of the repo when specifying the path? Is the `./` needed?

Comment (annakrystalli) [reply scripts/test-file-size.sh]: Do you mean, if you are running the tests locally?

Comment (annakrystalli) [reply scripts/test-file-size.sh]: Or do you mean to ensure paths are specified correctly in the script if it's edited?

**Comment by alexmorley** [comment]
> Or do you mean to ensure paths are specified correctly in the script if it's edited?

This one, just for clarify (it's a pretty minor thing).

On Thu, Mar 28, 2019 at 10:39 AM Anna Krystalli <notifications@github.com<mailto:notifications@github.com>> wrote:

@annakrystalli commented on this pull request.

________________________________

In scripts/test-file-size.sh<https://github.com/alan-turing-institute/the-turing-way/pull/354#discussion_r269942038>:

> @@ -0,0 +1,21 @@
+#! /bin/bash
+# add any files you want the test to ignore between the two ENDOFLIST placeholders.


Or do you mean to ensure paths are specified correctly in the script if it's edited?

**Comment by annakrystalli** [comment]
Gah!! I'm sorry I closed this by mistake!! 🤦‍♀️

**Comment by KirstieJane** [comment]
https://github.com/alan-turing-institute/the-turing-way/pull/354#issuecomment-477536903

Fixing this now!

**Comment by pherterich** [comment]
@annakrystalli all the other PRs now are yellow "Expected — Waiting for status to be reported" - Is Travis confused?

**Comment by ptheywood** [comment]
> @annakrystalli all the other PRs now are yellow "Expected — Waiting for status to be reported" - Is Travis confused?

Travis is currently having a bit of an outage, with a large backlog of builds to work through. 

https://www.traviscistatus.com/

**Comment by KirstieJane** [comment]
@annakrystalli - can this be merged? I'd like the test to be in the repo....

**Pull request merged**